### PR TITLE
Remove unnecessary command_buffer `begin/finish`

### PR DESCRIPTION
**Pull request by imiklos**
We can start record to the command_buffer at the next frame initialization and finish it before the submit.

### CODE REVIEW

File: webrender/src/device/gfx/device.rs [hunk 1]
```diff
@@ -471,7 +471,12 @@ impl<B: hal::Backend> Device<B> {
             }
             .expect("create_command_pool_typed failed");
             unsafe { hal_cp.reset() };
-            let cp = CommandPool::new(hal_cp);
+            let mut cp = CommandPool::new(hal_cp);
+            cp.acquire_command_buffer();
+            /// Start recording for the 1st frame
```
Comment (zakorgy) [inline webrender/src/device/gfx/device.rs]: Use `//` comments instead of doc comments, because this generates warnings and fails the CI.

Comment (imiklos) [reply webrender/src/device/gfx/device.rs]: Thank you! I've addressed the problem and rebased the PR to the master.

File: webrender/src/device/gfx/command.rs [hunk 2]
```diff
@@ -8,39 +8,36 @@ use hal::pool::RawCommandPool;
...
-            size: 0,
+            command_buffers: vec![],
         }
     }
 
-    pub(super) fn acquire_command_buffer(
-        &mut self,
-    ) -> &mut B::CommandBuffer {
-        if self.size >= self.command_buffers.len() {
+    pub(super) fn acquire_command_buffer(&mut self) {
```
Comment (zakorgy) [inline webrender/src/device/gfx/command.rs]: We could rename this to something like `init_command_buffer` or `create_command_buffer`.

File: webrender/src/device/gfx/command.rs [hunk 3]
```diff
@@ -8,39 +8,36 @@ use hal::pool::RawCommandPool;
...
-        self.size += 1;
-        &mut self.command_buffers[self.size - 1]
     }
 
     pub(super) fn command_buffers(&self) -> &[B::CommandBuffer] {
-        &self.command_buffers[0 .. self.size]
+        &self.command_buffers
+    }
+
+    pub fn get_command_buffer(&mut self) -> &mut B::CommandBuffer {
```
Comment (zakorgy) [inline webrender/src/device/gfx/command.rs]: The function name should tell us we receive a mutable borrow here, `command_buffer_mut` would be more suitable I think.

Comment (imiklos) [reply webrender/src/device/gfx/command.rs]: Thank you! I've renamed these functions.

**Pull request merged**